snippets/MoveOnScroll.js

```diff
@@ -0,0 +1,235 @@
+            return "";
+
+        const translateY = el.vertical * percent;
+        return percent ? "translateY(" + translateY + "px) " : "translateY(0px)";
+    }
+
+    function getScale(el, percent) {
+        if (el.scale === false || el.scale === null)
+            return "";
+
+        const [startScale, endScale] = el.scale;
+        let currentScale;
+
+        if (el.progressive)
+            currentScale = startScale + (endScale - startScale) * percent;
+        else
+            currentScale = percent ? endScale : startScale; //Percent == 1 => jump to endScale
+
+        return "scale(" + currentScale + ")";
+    }
+
+    function getOpacityValue(el, percent) {
+        if (el.opacity === false || el.opacity === null) return 1;
+
+        const [startOpacity, endOpacity] = el.opacity;
+        let currentOpacity;
+
+        if (el.progressive)
+            currentOpacity = startOpacity + (endOpacity - startOpacity) * percent;
+        else
+            currentOpacity = percent ? endOpacity : startOpacity; //Percent == 1 => jump to endOpacity
+
+        return currentOpacity;
+    }
+
+    function applyStyle(el, styleProperty, value) {
+        el.element.htmlElement.style[styleProperty] = value;
+    }
+
+    function arrangeElements(percent) {
+        elements.forEach(function (el, idx) {
+            if (!el.element.htmlElement)
+                return;
+
+            //percent calculated relative to element scroll interval. 
+            //In case of progressive: false, it return 1 (inside triggerScrollInterval) or 0 (outside)
+            const elementPercent = getElementPercent(el, percent);
+
+            //Move element to position
+            updatePosition(el, elementPercent);
+        });
+    }
+
+    function getElementPercent(el, percent) {
+        const [minTrigger, maxTrigger] = el.triggerScrollInterval;
+
+        if (el.progressive)
+            return remapRange(percent, [minTrigger, maxTrigger], [0, 1]);
+
+        const isInRange = percent >= minTrigger && percent <= maxTrigger;
+        if (isInRange)
+            return 1;
+
+        return 0;
+    }
+
+    //At screen change, reset elements positions
+    function resetElementsPositions() {
+        const percent = getScrollPercent();
+
+        toggleElementTransitions(false);
+        arrangeElements(percent);
+        toggleElementTransitions(true);
+    }
+
+    function toggleElementTransitions(shouldEnable) {
+        elements.forEach(function (el, idx) {
+            //Skip element if HTML element not yet loaded, or if progressive (transition not needed) 
+            // if (!el.element.htmlElement || el.progressive)
+            // return;
+
+            if (shouldEnable)
+                bnt.requestAnimFrame(function () { addElementTransition(el); });
+            else
+                removeElementTransition(el);
+        });
+    }
+
+    function getPlayerBounds(start, end) {
+        let percent = 0;
+        const getTeadsApi = () => {
```

Comment:
iti adaug niste spatii aici pt vizibilitate.
```suggestion

const getTeadsApi = () => {
```